org.eclipse.jdt.apt.core/src/org/eclipse/jdt/apt/core/internal/AptCompilationParticipant.java

```diff
@@ -208,6 +215,45 @@ public void processAnnotations(BuildContext[] allfiles) {
 		}
 	}
 
+	@Override
+	public URI[] getAnnotationProcessorPaths(IJavaProject project, boolean isTest) {
```

Comment:
The motivation is that the AbstractImageBuilder contains two compilation groups (MAIN and TEST), which invoke the CompilationParticipant separately. In the current experiment stage, it doesn't use this flag because I haven't encountered a use case that requires differentiating annotation processors for the test scope.

Reply:
I don't see any code in JDT that supports separate annotationProcessorPaths for MAIN vs TEST, only separate folders for generated sources. Am I missing anything?

Anyway this can be cleaned up later.